### PR TITLE
feat(builder): a right-click menu over the block verbs

### PR DESCRIPTION
**Pull request by mobeenabdullah**
Closes `finding:pb-canvas-has-no-context-menu` — founder-reported, and the first of the four canvas items in the order they chose (context menu → layers reordering → zoom → insert grid).

Right-clicking a block did nothing. Verified two ways before building: zero `onContextMenu` anywhere in `packages/*/src` (control: 9 `onDoubleClick` in the builder, so the search works), and in a browser, a real secondary click left `defaultPrevented` false with no `[role=menu]` in the DOM. A missing feature, not a broken one.

## What made this small

`@nextlyhq/ui` **already ships a `ContextMenu`**, built on `@radix-ui/react-context-menu` and exported from its index — ARIA roles, arrow-key navigation, typeahead, Escape, focus return, portalling. `packages/builder` already depends on it. Nothing here re-implements a menu.

## It derives; it does not decide

- **Which verbs, and whether each is available** — `toolbarActions`, the same call the floating bar makes. `builder-commands` already states the rule for the palette: asking a second time "is how a palette ends up offering a command that the button beside it shows as unavailable."
- **What each one runs** — `blockActionRunners`, lifted out of `builder-commands` and published, so the palette and this menu cannot bind the same label to different verbs. It is total over `ToolbarActionId`, so a verb added to the bar cannot reach a surface with nothing bound to it.
- **Labels** — the toolbar's, not the palette's. The palette deliberately says "Duplicate block" because a row is read in a list with no context; this menu is drawn *at* the block, like the bar.

An unavailable verb stays and carries its reason rather than disappearing — a menu that silently drops Delete answers "why can I not delete this" with nothing.

## Selection

The canvas moves the selection to the block under the pointer **on the contextmenu event itself**, before anything can open. A menu opened over one block while the selection sits on another aims its destructive verbs somewhere the author cannot see. A block already in the selection keeps it, so right-clicking one of several does not drop the rest. Over the background the canvas stops the event — a menu of block verbs there has no subject.

## Why the trigger generates no box

Radix's trigger renders a `span`; wrapped around the canvas that is an inline box holding a block one. `display: contents` leaves it in the DOM generating no box — events bubble along the tree, and Radix anchors to the pointer, not the trigger's rect. Keeping it *above* the canvas root also keeps the canvas's drag handlers from colliding with the pointer handlers Radix binds for touch long-press.

## Verified in a real browser

Playground, real CDP right-clicks (not synthetic dispatch — a dispatched `contextmenu` does **not** open it, which is its own reminder that programmatic events prove nothing here):

| case | result |
|---|---|
| right-click a block | menu opens at the pointer (offset dx=2, dy=0), block selected |
| items | `Select parent`, `Move up`, `Move down` (disabled — lone top-level block), `Duplicate`, `Delete` |
| **under canvas CSS `zoom` 0.6** | **identical offset dx=2, dy=0** |
| right-click the background | no menu |

The zoom row settles a question the research flagged and did not assert: the canvas scales with CSS `zoom`, where `clientWidth` reads 1017 logical while the rect reads 610 post-zoom. Radix anchors from the native event's viewport coordinates, so no mapping is needed.

## One limitation, measured and stated

**This menu is reachable by pointer only.** Shift+F10 with the canvas focused does not open it, and the structure says why: the focusable canvas region is an *ancestor* of the trigger (verified `section.contains(.nx-canvas) === true`), so a keyboard-invoked context event bubbles away from it.

It ships anyway because it takes nothing away — every verb in it is already reachable without a pointer through the keystrokes, the toolbar's roving tab stop and the command palette. The fix is not to raise the trigger, which would open a block menu over chrome that is not a block; it is to make blocks themselves focusable, which is `finding:pb-canvas-blocks-invisible-to-assistive-tech` and is the correct target for a context-menu key. The research node's recommendation of keyboard parity "from the canvas region" is recorded in the ledger as measured false.

## Tests

Eight new, each break-verified individually with the mutation being removal of the behaviour the test's *name* promises:

| mutation | fails |
|---|---|
| remove the canvas `onContextMenu` wiring | `selects the block under a secondary click…`, `withholds the event from above…` |
| always replace the selection | `keeps a selection that already holds the block` |
| stop *every* contextmenu event | `lets the event reach above when a block IS under it` |
| never render the menu content | all four menu cases |
| restate the list with palette wording | `offers the verbs the toolbar offers…` |
| bind every item to the wrong verb | `runs the verb the keystrokes run` |
| drop unavailable verbs instead of disabling | `keeps an unavailable verb…` |

No mutation kills `opens on a secondary click` alone, because every other case requires the menu to be open — a property of the dependency structure, not a gap.

The selection test drives Radix on `pointerup` with no preceding `pointerdown`: its item synthesises the click on pointer-up only when it saw no pointer-down, because a real browser delivers the click itself and jsdom delivers neither. A full down-then-up selects nothing, and so does `fireEvent.click`.

Four `BlocksField` test mocks gained `BlockContextMenu: passthrough` — their `BlockKeyboardActions` stub provides no verbs context, and the canvas renders inside this component, so a `nothing` stub would take the recorder out of the tree with it.

Green: `builder` 2336, `plugin-page-builder` 500, typecheck and lint clean. Changeset validated by path, with a deliberately broken copy confirming the checker rejects.

### CODE REVIEW

**Comment by mobeenabdullah** [comment]
@codex review

File: packages/builder/src/index.ts [hunk 1]
```diff
@@ -330,11 +330,26 @@ export {
...
+ * Assembled here rather than by the host for the reason `EditorCommandPalette`
+ * is: mounting it means reaching into the verbs context, deriving the list from
+ * `toolbarActions` on the right renders, and knowing it must sit inside
+ * `BlockKeyboardActions`. A host given the pieces would get all three right
+ * differently.
+ */
+export {
+  BlockContextMenu,
+  type BlockContextMenuProps,
+} from "./block-context-menu";
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/builder/src/index.ts]: Correct, and this is the one I should have caught myself — `tsup.config.ts` builds `src/shell.ts` WITH the banner and `tsup.server-safe.config.ts` builds `src/index.ts` without it, and that file's own header says why. Fixed in 59b3bc798: exported from `shell.ts` alone.

Verified in the ARTIFACT rather than the source, because that is where the claim lives:

```
with the export:     BlockContextMenu in dist/index.mjs = 2,  "use client" banner = 0
without the export:  BlockContextMenu in dist/index.mjs = 0
control (shell.mjs): BlockContextMenu = 2,  "use client" banner = 1
```

Worth noting there is no guard for this. `packages/ui` runs `check-client-directive` and `check-server-safe-artifacts` in its build; `packages/builder` runs neither, so nothing but review stood between this and the published entry. Filing that as its own task.

File: packages/builder/src/canvas.tsx [hunk 2]
```diff
@@ -529,7 +533,43 @@ function useCanvasPointer(
...
+   * of the author's selection at the moment they went looking for a verb.
+   *
+   * Over the canvas BACKGROUND the event is stopped instead. There is no node,
+   * so a menu of block verbs would have no subject; stopping it here rather
+   * than opening an empty one keeps that decision next to the hit test that
+   * establishes it.
+   */
+  const contextMenu = useCallback(
+    (event: React.MouseEvent<HTMLDivElement>) => {
+      if (isChrome(event.target)) return;
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/builder/src/canvas.tsx]: Confirmed in a real browser, and more directly than I expected. Fixed in 59b3bc798 — the chrome branch stops the event now, like the background branch.

Measuring it: right-clicking the visual centre of a Box put `document.elementFromPoint` on an `svg` whose `closest("[data-nx-chrome]")` was non-null and whose `closest("[data-nx-node]")` was **null** — the empty-container appender's "+", drawn over the block. So this is not an edge case; it is what you hit aiming at the middle of an empty container.

It also caught a bad test of mine. My first fixture appended the chrome button to the canvas ROOT, and it passed with the rule deleted — chrome on the background resolves to no node, so the `id === null` branch stopped it anyway and the fixture proved nothing. The fixture puts the chrome INSIDE the block now, which is the shape that reaches the trigger with an id already resolved, and removing `isChrome` fails it.

File: packages/builder/src/canvas.tsx [hunk 3]
```diff
@@ -529,7 +533,43 @@ function useCanvasPointer(
...
+      const id = nodeIdFromEvent(event.target);
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/builder/src/canvas.tsx]: Agreed, and this is the one that would have hurt most. Fixed in 59b3bc798.

Verified your premise: `grep -c contextmenu packages/builder/src/use-inline-text.ts` is 0, while the same file handles `pointerdown` and `click` — so nothing was keeping the gesture away from the trigger, and the block menu would have replaced spelling, selection and clipboard for a caret mid-sentence.

The canvas now treats an editable target as an editing gesture and stops the event. `contenteditable="false"` is excluded explicitly, since that marks a region the editor has deliberately made uneditable inside an editable one — a block again rather than text. `preventDefault` is still not called anywhere, so the browser's own menu appears wherever this one does not.

Covered by a unit test that fails with the rule removed. Stated plainly: that one is verified in jsdom, not in the browser — I verified the chrome and zoom cases with real clicks but did not drive an inline-editing session.

File: packages/builder/src/block-context-menu.tsx [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+        so this is the other route to the same state: a selection that was
+        cleared while the menu's own markup still existed.
+      */}
+      {selectedId === null ? null : (
+        <ContextMenuContent aria-label="Block actions">
+          {actions.map(action => (
+            <React.Fragment key={action.id}>
+              {DESTRUCTIVE.has(action.id) ? <ContextMenuSeparator /> : null}
+              <ContextMenuItem
+                disabled={!action.enabled}
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/builder/src/block-context-menu.tsx:231]: Right on both halves, and I had flagged my own uncertainty here in a comment instead of resolving it — which is not good enough. Fixed in 59b3bc798: the reason is visible text inside the item, not a `title`.

Both routes are closed exactly as you say: `menuItemBase` in the shared component carries `data-[disabled]:pointer-events-none`, so there is no hover, and Radix skips disabled items in keyboard navigation, so there is no focus either. The tooltip existed in the markup and reached nobody.

Verified in a browser on a locked block:

```
Select parent                          disabled  h=32
Move up / This block is locked.        disabled  h=50
Move down / This block is locked.      disabled  h=50
Duplicate                              enabled   h=32
Delete / This block is locked.         disabled  h=50
```

The 50-vs-32 heights are the second line actually laid out, not just present in the DOM. Putting it in the item's content also means it is part of the accessible name, so it is announced rather than only seen.

(`Duplicate` staying enabled on a locked block is deliberate and pre-existing — `keyboard-actions` documents that duplicating neither moves nor deletes the original.)

**Comment by mobeenabdullah** [comment]
@codex review

File: packages/builder/src/canvas.tsx [hunk 5]
```diff
@@ -945,6 +1020,7 @@ export function Canvas({
       data-nx-selected-id={selectedId ?? undefined}
       onClick={pointer.onClick}
       onDoubleClick={pointer.onDoubleClick}
+      onContextMenu={pointer.onContextMenu}
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/builder/src/canvas.tsx:1090]: Real, and the most serious one on this PR — Delete aimed at a block the author is not looking at. Fixed in 2f4db691d.

Verified both halves in the installed package rather than taking the mechanism on trust:

- `@radix-ui/react-context-menu` has the 700ms timer and its trigger's `onPointerDown` is `composeEventHandlers(props.onPointerDown, <its own>)`.
- `@radix-ui/primitive`'s `composeEventHandlers` runs the passed handler FIRST and skips its own when `event.defaultPrevented`.

So preventing the default in a handler on the trigger is a supported way to stop that path, not a trick.

Rather than filter twice, the decision is now published from the canvas as `contextMenuTargetOf(target)` and asked by both routes — the `contextmenu` path and the pointer-down path. Two spellings of three rejections is exactly how one route comes to accept what the other rejects. On touch and pen the press either moves the selection to the block underneath, or prevents the default; a mouse press is left alone, since its own event has already been filtered by the time Radix sees it.

Three tests, each break-verified with a distinct mutation: removing the handler fails the selection case and the rejection case; selecting on every pointer type fails the mouse control; rejecting without `preventDefault` fails only the rejection case.

The selection is asserted rather than the menu opening, deliberately — the opening is Radix's timer to run, and the SUBJECT is what this had to get right.

**Comment by mobeenabdullah** [comment]
@codex review

File: packages/builder/src/block-context-menu.tsx [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+   */
+  const select = editor.select;
+  const pressed = React.useCallback(
+    (event: React.PointerEvent<HTMLDivElement>) => {
+      if (event.pointerType === "mouse") return;
+      const id = contextMenuTargetOf(event.target);
+      if (id === null) {
+        // Nothing to act on. Preventing the default is what keeps Radix's
+        // long-press timer from opening a menu over chrome or over text.
+        event.preventDefault();
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/builder/src/block-context-menu.tsx]: Right, and my fix for the desktop case created this one. Fixed in 050e7400c.

`preventDefault` on `pointerdown` was the wrong instrument at the wrong moment: it fires at the start of every contact, long before anything knows whether it will become a long press, so an author tapping into a sentence lost caret placement and text selection.

Rejection moves to the canvas, which now withholds the press from anything above it with `stopPropagation` — that stops Radix's timer (its handler is on an ancestor, so it never runs) without touching what the browser does with the contact. Same instrument the canvas already uses for the `contextmenu` path, and it keeps one decision governing both routes rather than two.

Asserted as the default SURVIVING a touch press on a `contenteditable` target, which is the observable of that choice; it fails if the canvas cancels the press instead of withholding it.

File: packages/builder/src/block-context-menu.tsx [hunk 7]
```diff
@@ -0,0 +1,220 @@
...
+        return;
+      }
+      // Already chosen means already correct, and replacing would drop the
+      // rest of a multiple selection the press was meant to act on.
+      if (!selectedIds.includes(id)) select(id, "replace");
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/builder/src/block-context-menu.tsx]: Agreed. Fixed in 050e7400c — the press is noted and nothing more, and the selection moves when the menu OPENS, via `onOpenChange`, which is the first moment a long press is known to be one.

You are right about the asymmetry that made it a defect: Radix cancels its own timer on `pointermove`, but a selection already made is not part of that cancellation, so a finger-scroll would have retargeted the inspector and the toolbar with nothing to put it back.

This also exposed a bad test of mine, which is worth stating. The case named "a touch long-press" drove a `contextmenu` event — so it passed with the open handler deleted, because the canvas had already selected on the very event the test sent. It proved the route it was named after was working while never exercising it. It now runs Radix's real 700ms timer under fake timers and asserts both the menu opening and the subject, and it fails with either the press-recording or the open handler removed.

**Comment by mobeenabdullah** [comment]
@codex review

File: packages/builder/src/block-context-menu.tsx [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+    []
+  );
+  const opened = React.useCallback(
+    (open: boolean) => {
+      if (!open) return;
+      const id = pressedTarget.current;
+      // Already chosen means already correct — the mouse route selected on the
+      // context event — and replacing would drop the rest of a multiple
+      // selection the gesture was meant to act on.
+      if (id === null || selectedIds.includes(id)) return;
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/builder/src/block-context-menu.tsx]: Real, and the route I had not accounted for. Fixed in 862c79680.

You are right that a rendered block can be focusable on its own — `core/button` draws a native button — so a menu key arrives here as a context event with no press before it, and the remembered target was whatever the last press had been pointing at.

The target is now recorded on the **context event as well as** the press, so whichever gesture opens the menu is the one that answers for its subject, and it is consumed when the menu opens.

One honest note on coverage. The new case ("opens on the block the gesture named, not the one pressed before it") is break-verified by removing the `onContextMenu` recording — that is the half that actually fixes the reported defect. Removing the *clear* alone fails nothing, because the recording already refreshes the value; I kept it because consume-on-open is the correct model for a one-shot handoff and it keeps a future opening route from silently inheriting a stale answer, but I am not claiming it is independently observable.

File: packages/builder/src/block-context-menu.tsx [hunk 9]
```diff
@@ -0,0 +1,232 @@
...
+      select(id, "replace");
+    },
+    [select, selectedIds]
+  );
+
+  return (
+    <ContextMenu onOpenChange={opened}>
+      <ContextMenuTrigger asChild>
+        <div style={TRANSPARENT_WRAPPER} onPointerDown={notePress}>
+          {children}
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/builder/src/block-context-menu.tsx:216]: Confirmed, including your point about my test. Fixed in 862c79680.

You are right that the old case only checked `defaultPrevented` and would have stayed green with the callout disabled — it asserted the thing I had just changed rather than the thing an author would lose.

The stylesheet restores it on the editable subtree alone: `.nx-canvas [contenteditable]:not([contenteditable="false"]) { -webkit-touch-callout: default; }`. Scoped that way so the block menu keeps the whole of the canvas that is not text, and `contenteditable="false"` is excluded for the same reason the canvas excludes it — that marks a region deliberately made uneditable inside an editable one, which is a block again.

Asserted against the SOURCE stylesheet, in `builder-chrome-attributes.test.ts` beside the existing selector checks, because no render test can see it: jsdom computes no inherited `-webkit-` property, and the suppression comes from a library's inline style rather than from anything in this package. Break-verified by replacing the declaration — the case fails.

**Comment by mobeenabdullah** [comment]
@codex review

File: packages/builder/src/block-context-menu.tsx [hunk 10]
```diff
@@ -0,0 +1,232 @@
...
+ * focused does NOT open it, and the structure says why — the focusable canvas
+ * region is an ANCESTOR of this trigger, so a keyboard-invoked context event
+ * originating there bubbles away from it rather than into it.
+ *
+ * It is shipped anyway because it takes nothing away. Every verb here is
+ * already reachable without a pointer, through the keystrokes, the block
+ * toolbar's roving tab stop, and the command palette — this is a fourth route
+ * to them, not the only one. The fix is not to move this trigger up, which
+ * would open a block menu over chrome that is not a block; it is to make the
+ * blocks themselves focusable, which is a separate defect already filed and is
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/builder/src/block-context-menu.tsx]: Correct, and this is my own standing rule that I broke — a comment describes the code, never where something is tracked. Fixed in 862c79680 (the reply is late; the fix went out with an earlier push and I lost the thread while the repo-wide push gate was blocking).

The sentence now says what the limitation IS and where a keyboard gesture would have to land, with no reference to anything outside the file. I also swept the whole diff for the same shape — `already filed`, `this PR`, `follow-up`, `TODO`, reviewer names — and the only remaining matches are false positives ("this press writes THIS field's value", "reported by the canvas").

File: packages/builder/src/builder-commands.ts [hunk 11]
```diff
@@ -101,6 +101,31 @@ export const EDITOR_GROUP = "Editor";
  * searched rather than aimed at, and a row that matches the search and then
  * refuses to run is worse there than one that was never offered.
  */
+/**
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/builder/src/builder-commands.ts:95]: Confirmed in the emitted types, which is where you looked and where it matters. Fixed in 862c79680.

I had inserted `blockActionRunners` between the palette's docblock and `builderCommands`, so the published declaration lost its documentation entirely — `dist/index.d.ts` showed the comment above the new helper and `builderCommands` bare. The helper and its own docblock moved above the palette's, and the emitted `.d.ts` now carries the palette docs immediately before `declare function builderCommands`.

Verified by rebuilding and reading the artifact rather than the source, since the source order was the thing that misled me in the first place.

File: packages/builder/src/canvas.tsx [hunk 12]
```diff
@@ -89,6 +89,47 @@ export const SELECTED_ATTRIBUTE = "data-nx-selected";
...
+ * that filtered one path and not the other would offer a block's verbs for a
+ * press aimed at chrome, or — worse — offer the PREVIOUS selection's verbs,
+ * with Delete among them, for a press on a block that was never selected.
+ *
+ * So the decision lives here once and both callers ask it, rather than each
+ * spelling out three rejections and drifting.
+ */
+export function contextMenuTargetOf(target: EventTarget | null): string | null {
+  if (isEditableTarget(target) || isChrome(target)) return null;
+  return nodeIdFromEvent(target);
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/builder/src/canvas.tsx]: Real, and the code already claimed the rule it was not enforcing — `nodeIdFromEvent`'s own comment says "Inside THIS canvas" while checking only that SOME canvas is an ancestor. Fixed in 5640243d9.

`contextMenuTargetOf` now takes the handling root and refuses an owner whose nearest canvas is not it. The canvas passes `event.currentTarget`; the menu passes the outermost canvas beneath its own wrapper, so a nested one is deeper and is not what it resolves.

That is enough on its own, because the outer canvas stops the event before the trigger above it can see it — the inner canvas accepts its own node and lets it bubble, and the outer one now rejects it and stops there.

Break-verified: relaxing the comparison back to `!== null` fails the new case, which builds a nested `.nx-canvas` with its own `[data-nx-node]` inside an outer block and asserts no menu opens and the foreign id is never selected.

File: .changeset/canvas-right-click-menu.md [hunk 13]
```diff
@@ -0,0 +1,35 @@
...
+"@nextlyhq/module-specifiers": patch
+---
+
+Right-clicking a block on the canvas opens a menu.
+
+Nothing happened before: the editor had no context menu anywhere, so an author looking for the block's actions where every comparable builder puts them found nothing. The menu offers the same verbs the floating toolbar offers, in the same order, with the same availability and the same reasons — it is another way to reach them rather than a second list that can disagree. A verb that is unavailable stays visible and says why, because a menu that silently drops Delete answers "why can I not delete this" with nothing at all.
+
+Right-clicking a block also selects it first, so the menu always acts on the block under the pointer. Right-clicking one of several selected blocks keeps that selection instead of replacing it.
+
+It is reached by pointer only for now. Every verb in it is already reachable without a pointer through the keystrokes, the block toolbar and the command palette, so this adds a route rather than becoming the only one.
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply .changeset/canvas-right-click-menu.md]: You are right, and I had it backwards in a way worth being explicit about: I asserted a limitation I could not test, and the assertion understated the feature.

I went to measure your claim and found the instrument cannot produce the event at all. With a capture listener on `document`, a synthesised `Shift+F10` through CDP yields **contextmenu events seen: 0**. So my original "Shift+F10 does not open it" was not a measurement of the product — it was silence from a tool that never fires the event, and I wrote it up as a finding about the code.

Corrected in 5640243d9, and split the way you describe:

- a block drawing a **focusable control** reaches this by the same path a right-click does, which IS verified — `core/button` renders an anchor carrying the node marker itself, so focus lands inside the trigger and the event bubbles through the canvas's selection and the wrapper's recording exactly as a pointer event does
- focus on the **canvas region** does not, and cannot: that region is an ancestor of the trigger, so its event travels away from it, and there is no block under it to act on either way

Both the module comment and the changeset now say this, and the keyboard half is explicitly marked as reasoned rather than measured, with the reason the tooling cannot settle it.

File: packages/builder/src/styles/builder-chrome.css [hunk 14]
```diff
@@ -1995,3 +1995,23 @@
...
+ * press there, so a long press on text would open neither menu: the platform's
+ * spelling, selection and clipboard callout suppressed, and nothing put in its
+ * place. Restored on the editable subtree only, so the block menu keeps the
+ * whole of the canvas that is not text.
+ *
+ * `[contenteditable="false"]` is excluded for the reason the canvas excludes
+ * it: it marks a region deliberately made uneditable inside an editable one,
+ * which is a block again rather than text.
+ */
+.nx-canvas [contenteditable]:not([contenteditable="false"]) {
```
[the remark this answers is not in the document]
Comment (mobeenabdullah) [reply packages/builder/src/styles/builder-chrome.css]: Fixed in 5640243d9. The negated selector is now two plain rules that say the same thing:

```css
.nx-canvas [contenteditable]        { -webkit-touch-callout: default; }
.nx-canvas [contenteditable="false"] { -webkit-touch-callout: none; }
```

The second is not merely restoring the default — a region the editor has deliberately made uneditable inside an editable one is a block again, so the block menu is the right menu there and the suppression should stand. Splitting it made that intent explicit rather than hiding it inside a `:not()`.

The stylesheet test asserts both halves now, and break-verified: removing the suppression fails it.

**Comment by mobeenabdullah** [comment]
@codex review
